packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -181,46 +194,56 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
 
     // Round down to the nearest ohlc period so the queries captures the OHLC of the period *before* this earliest
     // timestamp (because the close of that OHLC may be relevant).
-    const earliestHistoricalTimestamp = Math.floor((currentTime - this.lookback) / this.ohlcPeriod) * this.ohlcPeriod;
+    const earliestHistoricalTimestamp =
+      Math.floor((currentTime - (this.lookback + this.twapLength)) / this.ohlcPeriod) * this.ohlcPeriod;
+
+    const newHistoricalPricePeriods = await this._getOhlcPricePeriods(earliestHistoricalTimestamp, currentTime);
+    const newPrice = this.twapLength
+      ? this._computeTwap(currentTime, newHistoricalPricePeriods)
+      : await this._getImmediatePrice();
+
+    // 5. Store results.
+    this.currentPrice = newPrice;
+    this.historicalPricePeriods = newHistoricalPricePeriods;
+    this.lastUpdateTime = currentTime;
+  }
 
-    // 1. Construct URLs.
+  async _getImmediatePrice() {
```

Comment:
Is the idea eventually to make getCurrentPrice just return _getImmediatePrice()? I.e. skip the `update()` process

Reply:
Not exactly. `_getImmediatePrice` is only used when twapLength === 0. So if we were to remove `update`, `getCurrentPrice` would be a ternary that either returns `_getImmediatePrice` or `_computeTwap` depending on twapLength.